src/grammar_tester/parsevaluate.py

```diff
@@ -33,70 +43,188 @@ def load_ull_file(filename):
 
             line_count += 1
 
-        parses.append(parse)
+        index += 1
+        # parses.append(parse)
 
     return parses
 
 
-def eval_parses(test_parses: list, ref_parses: list, verbose: bool, ofile=sys.stdout) -> ParseQuality:
+def save_parses(sentence_parses: List[Tuple[str, set]], file_name: str, options: int) -> None:
+    """
+    Print parses to file (for sequential and random eval methods)
+
+    :param sentence_parses:     Parse info structure.
+    :param file_name:           Path to file to save parses to.
+    :param options:             Parse options bit mask.
+    :return:                    None.
+    """
+    print("writing parses file to '{}'".format(file_name))
+
+    with open(file_name, 'w') as fo:
+        for sent in sentence_parses:
+            print_output(["###LEFT-WALL###"] + (sent[0].strip()).split(), list(sent[1]), options, fo)
+
+    print("Finished writing parses file")
+
+
+def make_sequential(sentences: Union[List[Tuple[str, set]],List[str]], options: int, **kwargs) -> List[Tuple[str, set]]:
+    """
+    Make sequential parses (each word simply linked to the next one),
+    to be used as baseline
+
+    :param sentences:       List of either tuples of sentence and set of links in case of .ull input file format
+                            or strings in case of text input file format.
+    :param options:         Integer representing parse options bit masks.
+    :return:                List of parses (tuples of sentence and set of links)
+    """
+    sequential_parses = []
+
+    is_ull = True if isinstance(sentences, list) and isinstance(sentences[0], tuple) else \
+        False if isinstance(sentences, list) and isinstance(sentences[0], str) else None
+
+    if is_ull is None:
+        raise ValueError(f"The first argument should be either List[Tuple[str, set] or List[str], "
+                         f"but it is {sentences.__class__.__name__.capitalize()}"
+                         f"[{sentences[0].__class__.__name__.capitalize()}]")
+
+    for sp in sentences:
+        sentence = sp[0] if is_ull else sp
+        links = {(0, 1)}
+
+        for i in range(1, len(sentence.split())):
+            links.add((i, i + 1))
+
+        sequential_parses.append((sentence.strip(), links))
+
+    return sequential_parses
+
+
+def make_random(sentences: Union[List[Tuple[str, set]],List[str]], options: int, **kwargs) -> List[Tuple[str, set]]:
+    """
+    Make random parses (from LG-parser "any"), to use as baseline
+
+    :param sentences:       List of either tuples of sentence and set of links in case of .ull input file format
+                            or strings in case of text input file format.
+    :param options:         Integer representing parse options bit masks.
+    :return:                List of parses (tuples of sentence and set of links)
+    """
+    any_dict = Dictionary('any') # Opens dictionary only once
```

Comment:
These could use the API in lgapiparser.py, right?

Reply:
If you mean using "any"  dictionary within lgapiparser.py - yes. Actualy, LGAPIParser and LGInprocParser are having the same base class, so are interchangeble. LG python API had problems in previous versions, that's why LGInprocParser were created. LGInprocParser is used by default by GrammarTester. One common test for all LG parsers is needed to make sure they are consistent. I am planning to make it shortly. Now LGInprocParser is used for all parsing activities.